lib/IRGen/GenDecl.cpp

```diff
@@ -1252,12 +1252,6 @@ static bool isLazilyEmittedFunction(SILFunction &f, SILModule &m) {
   if (f.getDynamicallyReplacedFunction())
     return false;
 
-  // Needed by lldb to print global variables which are propagated by the
-  // mandatory GlobalOpt.
-  if (m.getOptions().OptMode == OptimizationMode::NoOptimization &&
-      f.isGlobalInit())
-    return false;
-
```

Comment:
Wouldn't you need to call `shouldBePreservedForDebugger()` here, too? So that such functions will not get emitted lazily.

Reply:
I added the call to `shouldBePreservedForDebugger`  to `isPossiblyUsedExternally`, which is called by `isLazilyEmittedFunction`. Do you think that's too broad?